### ISSUE TITLE
API use case for dictionaries

### ISSUE THREAD

**Issue by drdhaval2785** [sanskrit-lexicon/COLOGNE#329]
https://ashtadhyayi.com/dhatu/01.0001
Click on भवन्ति
This would open a pop up giving step by step derivation for the word भवन्ति.
This data of derivation comes from https://api.sanskritworld.in/ .

Can we prepare some APIs which would be able to provide the same for displaying Cologne dictionary data to similar pages?
Let us say some website has 'rAmAya' on the site.
When I click on 'rAmAya', I should get data about 'rAma' on all dictionaries. 

It would be a three step process I guess.
1. Parse Sandhi
2. Remove the case endings
3. Search the headword in dictionaries.

**Comment by gasyoun** [comment]
For now even an API that can return the meaning of `bhū` for `bhū` would 
be great. Let `bhū` for `भवन्ति` @funderburkjim come as the next step.

**Comment by drdhaval2785** [comment]
> For now even an API that can return the meaning of bhū for bhū would be great. 

I completely agree. After closing servepdf API, this is the API I would request Jim to handle. getword.php is a close candidate for this purpose.

**Comment by gasyoun** [comment]
**Question type:** editorial-policy / documentation

**Evidence found:**
The `csl-apidev` repository now has documented API families available at [doc/readme.md](https://github.com/sanskrit-lexicon/csl-apidev/blob/master/doc/readme.md):

1. **Classic PHP endpoints** (`getword.php`, `listview`, `getsuggest`) — already handle the base case requested here: look up a headword and return dictionary data. Example: `getword.php?dict=mw&key=rAma` returns all MW entries for 'rāma'.
2. **Salt API** (C-SALT-compatible JSON/GraphQL) — newer contract documented in [doc/salt_entries.md](https://github.com/sanskrit-lexicon/csl-apidev/blob/master/doc/salt_entries.md), [doc/salt_graphql.md](https://github.com/sanskrit-lexicon/csl-apidev/blob/master/doc/salt_graphql.md). Supports batch lookup, structured JSON fields, GraphQL selection.
3. **Clean-URL permalinks** — human-friendly links for citations (e.g. `/MW/rAma`).

The three-step flow described in the issue (Sandhi parse → remove case endings → dictionary lookup) maps to: external sandhi/morphology tool → `getword.php` or Salt API for the lookup step. The Cologne side of step 3 is already implemented.

**Assessment:**
Answerable: The basic use case (look up a headword across dictionaries via API) is already implemented via `getword.php`. The Salt API adds structured JSON for programmatic use. Steps 1–2 (Sandhi splitting, morphological analysis) remain external to CDSL infrastructure; ashtadhyayi.com uses https://api.sanskritworld.in/ for that, which could be combined with the CDSL `getword` endpoint on the client side.

**Suggested next step:** Link this issue to [csl-apidev](https://github.com/sanskrit-lexicon/csl-apidev) in the issue body and note that the dictionary lookup layer (step 3) is already available via `getword.php`; a third-party caller combining a morphology API with `getword` can achieve the full flow today without changes to CDSL infrastructure.
